### PR TITLE
Rentify 16 create history of clicks

### PR DESCRIPTION
**Pull request by maria-markova**
Added history of clicks.
Added event scheduler for deleting views table every 7 days.

### CODE REVIEW

File: back-end/src/main/java/com/devminds/rentify/controller/ViewsController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/rentify/views")
+public class ViewsController {
+    private final ViewsService viewsService;
+
+    @Autowired
+    public ViewsController(ViewsService viewsService) {
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/controller/ViewsController.java:23]: Use `@RequiredArgsConstructor `instead.

File: back-end/src/main/java/com/devminds/rentify/service/ViewsService.java [hunk 2]
```diff
@@ -2,21 +2,26 @@
...
-public class HistoryService {
+public class ViewsService {
+    private static final String EMAIL_NOT_FOUND_MESSAGE = "Email %s was not found.";
     private final HistoryRepository historyRepository;
+    private final UserRepository userRepository;
     private final ModelMapper modelMapper;
 
     @Autowired
-    public HistoryService(HistoryRepository historyRepository, ModelMapper modelMapper) {
+    public ViewsService(HistoryRepository historyRepository, UserRepository userRepository, ModelMapper modelMapper) {
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/service/ViewsService.java:22]: You can use lombok here as well.

File: back-end/src/main/resources/db/migration/V8__create_scheduler_for_clearing_views_table.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+SET GLOBAL event_scheduler = ON;
+
+DELIMITER //
+
+CREATE PROCEDURE delete_views()
+BEGIN
+    DELETE FROM history;
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/resources/db/migration/V8__create_scheduler_for_clearing_views_table.sql:7]: You shouldn't clear all views just those older than 7 days, I think.

File: front-end/.gitignore [hunk 4]
```diff
@@ -5,12 +5,8 @@
 /.pnp
 .pnp.js
 
-#images
-*.jpg
```
Comment (KalinVasilev-at-blankfactor) [inline front-end/.gitignore:9]: Why did you remove these from the .gitignore?

Comment (maria-markova) [reply front-end/.gitignore:9]: if someone wants to clone this project, they won't be able to run it, because of some images we use in the FE like logo and some other default pictures.

File: back-end/src/main/java/com/devminds/rentify/controller/ViewsController.java (hunk 1 above)
Comment (maria-markova) [reply back-end/src/main/java/com/devminds/rentify/controller/ViewsController.java:23]: done.

File: back-end/src/main/java/com/devminds/rentify/service/ViewsService.java (hunk 2 above)
Comment (maria-markova) [reply back-end/src/main/java/com/devminds/rentify/service/ViewsService.java:22]: done

File: front-end/.gitignore (hunk 4 above)
Comment (maria-markova) [reply front-end/.gitignore:9]: fixed.

**Review by KalinVasilev-at-blankfactor — APPROVED** [review]

**Review by ValentinTsonevskii — APPROVED** [review]
